app/controllers/api/receipts_controller.rb

```diff
@@ -1,16 +1,18 @@
-class Api::TransactionsController < ApplicationController
+class Api::ReceiptsController < ApplicationController
   respond_to :json
   def show
   end
 
   def create
-    transaction = Transaction.new(transaction_params)
+    transaction = Receipt.new(receipt_params)
     if transaction.save!
+      transaction.request.open = false
+      transaction.request.save!
```

Comment:
for consistency should all instances of 'transaction' be renamed to 'receipt'? or is there a reason we're still calling them transactions?